src/SourceBuild/content/repo-projects/Directory.Build.targets

```diff
@@ -138,8 +139,14 @@
       <_BuildSources Include="@(_CommonBuildSources)" />
       <_BuildSources Include="$(NetSdkSupportingFeedName)"
                      Condition="'$(AddNetSdkSupportingFeed)' == 'true'" />
+      <_OfflineBuildSectionsToRemove Include="auditSources" />
     </ItemGroup>
 
+    <RemoveSectionsFromNugetConfig
+      NuGetConfigFile="$(NuGetConfigFile)"
+      SectionsToRemove="@(_OfflineBuildSectionsToRemove)"
+      Condition="'$(DotNetBuildWithOnlineFeeds)' != 'true'" />
```

Comment:
Seeing this offline condition does make the implementation smell a bit in that it removes the auditSources entirely rather than just only online sources.  It may never be an issue but it does give make me question.  Why not utilize the existing RemoveInternetSources task?  This current logic would handle removing the internet sources from auditSources with some minor tweaking.

Reply:
Yeah, that's a good point and a variation of the alternative solution proposed in the issue: https://github.com/dotnet/source-build/issues/4553

In the past we've considered removing some other sections from nuget.config, so having a task to do that seemed natural, but I see a value in reusing existing task.

Reply:
Abandoning this PR in favor of https://github.com/dotnet/sdk/pull/42732, as most of the comments are not applicable anymore.